### ISSUE TITLE
CONFIGURE cran nosuchpackage silently fails though appears successful

### ISSUE THREAD

**Issue by gjwgit** [mlhubber/mlhub#71]
For the audit repository, the MLHUB.yaml contains a cran dependency on 'nosuchpackage' and reports that it is installing the latest version and appears to succeed!

```
*** Installing latest version R package 'nosuchpackgr' from CRAN into
'/home/gjw/.mlhub/audit/R' ...

*** Installing latest version R package 'rpart' from CRAN into
'/home/gjw/.mlhub/audit/R' ...
```

I noticed this because I had included a misspelt package name in the dependencies.

**Comment by simonzhaoms** [comment]
In `mlhub/scripts/dep/r.R` which is used for installing required R packages:

```r
  already <- installed.packages()[,"Package"]
  already_ver <- installed.packages()[,"Version"]
  for (pkg in name_pkgs)
  {
    if (pkg %in% already)
    {
      latest_ver <- old.packages(instPkgs=installed.packages()[pkg, , drop=FALSE])[, 'ReposVer']
      if (is.null(latest_ver))
      {
        cat(sprintf("\n*** The R package '%s' is already the latest version.\n", pkg))
        next
      }
    }

    cat(sprintf("\n*** Installing latest version R package '%s' from CRAN into\n'%s' ...\n", pkg, .libPaths()[1]))
    install.packages(pkg, lib=lib)
  }
```

At the last second line `install.packages(pkg, lib=lib)`, if the `pkg` is invalid name, `install.packages` will give:

```r
$ R
> install.packages('nosuchpackage')
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning in install.packages("nosuchpackage") :
  'lib = "/usr/local/lib/R/site-library"' is not writable
Would you like to use a personal library instead? (yes/No/cancel) yes
Would you like to create a personal library
‘~/R/x86_64-pc-linux-gnu-library/3.5’
to install packages into? (yes/No/cancel) yes
Warning message:
package ‘nosuchpackage’ is not available (for R version 3.5.1)
```

We need to figure out how to capture the warning from `install.packages()`

**Comment by gjwgit** [comment]
Could use available.packages()?

**Comment by simonzhaoms** [comment]
Solved in commit d0691be
